pkg/extract/freebsd/freebsd.go

```diff
@@ -55,10 +66,144 @@ func Extract(args string, opts ...Option) error {
 			return nil
 		}
 
+		f, err := os.Open(path)
+		if err != nil {
+			return errors.Wrapf(err, "open %s", path)
+		}
+		defer f.Close()
+
+		var fetched freebsd.Vuln
+		if err := json.NewDecoder(f).Decode(&fetched); err != nil {
+			return errors.Wrapf(err, "decode %s", path)
+		}
+
+		extracted := extract(fetched)
+		if err := util.Write(filepath.Join(options.dir, fmt.Sprintf("%s.json", extracted.ID)), extracted); err != nil {
+			return errors.Wrapf(err, "write %s", filepath.Join(options.dir, fmt.Sprintf("%s.json", extracted.ID)))
+		}
+
 		return nil
 	}); err != nil {
 		return errors.Wrapf(err, "walk %s", args)
 	}
 
 	return nil
 }
+
+func extract(fetched freebsd.Vuln) types.Data {
+	if fetched.Cancelled != nil {
+		return types.Data{
+			ID:         fetched.Vid,
+			DataSource: source.FreeBSD,
+		}
+	}
+
+	reflen := len(fetched.References.URL)
+	reflen += len(fetched.References.FreebsdSA)
+	reflen += len(fetched.References.FreebsdPR)
+	reflen += len(fetched.References.Mlist)
+	reflen += len(fetched.References.BID)
+	reflen += len(fetched.References.CertSA)
+	reflen += len(fetched.References.CertVU)
+	reflen += len(fetched.References.USCertTA)
+
+	rs := make([]reference.Reference, 0, reflen)
```

Comment:
```suggestion
	rs := make([]reference.Reference, 0, len(fetched.References.URL)+len(fetched.References.FreebsdSA)+len(fetched.References.FreebsdPR)+len(fetched.References.Mlist)+len(fetched.References.BID)+len(fetched.References.CertSA)+len(fetched.References.CertVU)+len(fetched.References.USCertSA)+len(fetched.References.USCertTA))
```

Reply:
vuxml URL with Vid is added.

I want to discuss about the loooong sum line.

Reply:
How about adding line breaks one by one like this?
Personally, I don't really want to create a `reflen`.
https://github.com/MaineK00n/vuls-data-update/pull/174/files#r1580441584

Reply:
Understood! Applied your second suggestion! Thanks!!!